runtime/src/iree/hal/local/executable_library_util.c

```diff
@@ -113,17 +124,34 @@ iree_zone_id_t iree_hal_executable_library_call_zone_begin(
 
   const char* source_file = NULL;
   size_t source_file_length = 0;
-  uint32_t source_line;
-  if (library->exports.src_locs != NULL) {
-    // We have source location data, so use it.
-    source_file = library->exports.src_locs[ordinal].path;
-    source_file_length = library->exports.src_locs[ordinal].path_length;
-    source_line = library->exports.src_locs[ordinal].line;
-  } else {
-    // No source location data, so make do with what we have.
-    source_file = executable_identifier.data;
-    source_file_length = executable_identifier.size;
-    source_line = ordinal;
+  uint32_t source_line = 0;
+  if (library->exports.stage_locations != NULL) {
+    for (uint32_t i = 0; i < library->exports.stage_locations->count; ++i) {
+      // TODO(benvanik): a way to select what location is chosen. For now we
+      // just pick the first one.
+      // const char* name = library->exports.stage_locations->names[i];
+      const iree_hal_executable_source_location_v0_t* location =
+          &library->exports.stage_locations->locations[i];
```

Comment:
Keeping commented out code? Or cleanup?

Reply:
Could we tell developers to recompile the runtime with a different `#define` if they want to switch between `0_source`, `1_configured`, and `2_translated`?

Reply:
(Are all of those captured in the .vmfb files, but only one is sent to Tracy?)

Reply:
there as part of the todo, indented